### ISSUE TITLE
Flatpak or AppImage package

### ISSUE THREAD

**Issue by 13hannes11** [GraphiteEditor/Graphite#584]
It would be great to have an installable editor, for example as Flatpak or AppImage.

**Comment by Keavon** [comment]
Would a progressive web app (PWA) be sufficient? We should probably set up the manifest for that to work better by default. My concern with an installable editor that isn't a PWA would be the burden to provide updates and deal with that infrastructure. We'll have to handle that in the long-term when we have a desktop app, but I'm worried it will distract from the pace of development right now if we have to maintain more infrastructure. If you have any thoughts about how this might be simpler or more important than I'm realizing, I am all ears so please let me know!

**Comment by 13hannes11** [comment]
I think something like https://tauri.app could be used but it's understandable that as of now a purely web based app is easier to maintain. And unfortunately I neither have any experience with tauri nor do I know much about the requirements for Graphite.

For the sake of getting users it might be nice though to at least provide some wrapper that allows a "local" install while just pointing to the website wrapped in a browser window.

**Comment by Keavon** [comment]
Yeah, we've looked at Tauri and it looks like a promising project for the future, but isn't really ready yet. I think PWAs are the way to go at the moment, so we should certainly set up the manifest correctly so the browser prompts for that with the correct metadata. And then we can hopefully do whatever is feasible for a desktop version that doesn't require a big maintenance burden. (So if anyone is reading this in the future, please feel free to suggest ways that this could be achieved without much dev overhead.)

**Comment by alerque** [comment]
For other people hitting this open issue looking for dekstop info, it look like #636 implemented a Tauri based desktop wrapper. This issue for containerized Flatpak/AppIamage may be relevant for folks that want those packaging formats, but if you just want a desktop something the discussion in this issue is old news.

That being said build instructions are pretty sparse. I can get the web stuff to build to wasm and a desktop binary to build, but so far getting all the assets combined so the desktop build finds the static web and wasm assets it needs is not going well...

**Comment by Keavon** [comment]
We're working towards getting official Tauri builds supported in the next few months. We can look into which package managers to offer it through after we have downloads on our website. We're blocked on https://github.com/tauri-apps/tauri/issues/9220 for Linux though.

**Comment by BearzRobotics** [comment]
A flatpak would be as an official distribution package as most distros have flatpak in their repos. Then you can ship one blessed package. 

As for myself, I found the app from the interview with Developer Voice, and can't build it from source easily because libsoup 2.4 not being the in the chimera linux repo. Something in your deps relies on https://crates.io/crates/soup2-sys and it hasn't been updated in years to bind against newer versions of libsoup.

**Comment by Keavon** [comment]
@BearzRobotics could you please test it with the branch in #2362? That has our recently ongoing work to upgrade to Tauri v2 and revive the unmaintained Tauri codebase that has been sitting around and decaying in master. The latest status on that is an issue with compositing webview and WGPU textures in the same window without having to fall back to X11 from Wayland, which requires upstream development in Tauri. I think the current plan is to start with the X11 fallback then hope to find a contributor who can assist in fixing that blocker in Tauri for us. Or perhaps you can help get involved if you have experience in that area of desktop native API programming?

**Comment by BearzRobotics** [comment]
@Keavon  I did a fresh git clone to the branch [2362](https://github.com/GraphiteEditor/Graphite/pull/2362) 

git clone -b tauri-decouple-executor https://github.com/GraphiteEditor/Graphite.git
cd Graphite
cargo build

It built!

While I would love to help, I don't have any real experience programming.

**Comment by Keavon** [comment]
Thanks for testing it out. I'm glad to see it works on your end. The more systems to test with, the better :)

Feel free to subscribe to that PR for updates as we make progress towards landing it in master.

**Comment by koffydrop** [comment]
Hey, chiming in to say i got it to build and run with a couple minor issues on my nixos system! (x86_64, somewhat beefy specs)

Namely:
- cargo-about isn't included in the flake.nix file, thus `npm run build` wouldn't work until i added it.
- tauri.conf.json has `app.windows[0].decorations = false` and on KDE at least that makes the window not draggable or resizable without holding the super key, though i presume it's something to be polished in the future.

Other the that, and lacking clear build instructions, everything seemed to work. Looking forward to the future of Graphite!

**Comment by Keavon** [comment]
@koffydrop could you please post that in the PR?

**Comment by Keavon** [comment]
Just an update for those following, #2535 is now the tracking issue for the overall desktop app initiative. Move your 👍 from this issue to that one if you were upvoting the desktop app in general (rather than specifically distribution through Linux packaging formats).

**Comment by Keavon** [comment]
Related reading for future reference: https://jangafx.com/insights/linux-binary-compatibility

**Comment by ttang4299** [comment]
I'll vouch for distributing the Flatpak build through Flathub as well. As someone who is still rather new to Linux, Flatpaks are not only an easy way to discover apps, but also an easy way to keep various apps updated on Linux via a single UI, especially with Flathub.

**Comment by Keavon** [comment]
Current list of planned Linux app stores:
- https://flathub.org/
- https://snapcraft.io/store
- https://archlinux.org/packages/
- https://search.nixos.org/packages
- https://store.steampowered.com/

**Comment by alerque** [comment]
Not all of those are "app stores" *per se*.

Flathub, Snapcraft, and Steam **are** of course app stores and as an app developer you can target builds for those platforms.

Arch Linux and Nix are not for various reasons.

* The official Arch Linux package repository is managed by Arch package maintainers and requires the distro to *target you*, not the other way around. This is true of most Linux distros — you'll need people on the inside loop of the packaging system to get packages into most systems.

    That being said I am personally an Arch Linux package maintainer, already have [graphite-editor-git](https://aur.archlinux.org/packages/graphite-editor-git) in the AUR, and will be more than willing to adopt Graphite as an official package as soon as it has a tagged release (probably even if that is an early 0.0.1 or other early cycle release not considered production ready as long as the release process itself looks semi stable.

* Nix is similar in that it is a distro, but they do accept outside contributions to packages without any particular bars you'll have to cross about notoriety as long as someone from the community sounds like they will be pitching in for updates.

    However you do not have to and **should not wait** for an official Nix package in the nix-pkgs repository. You **should** have a Flake in this repository no that anybody with either Nix OS or any distro with a `nix` package can directly run your app. This can be done any time and doesn't need a tagged release to get started with, and working out the details of the build in the flake will be hugely beneficial to other potential distro packagers, eventually nixpkgs, your own development team to confirm dependencies in a hermetically sound way, and of course to users to be run in a predictable way as `nix run github:GraphiteEditor/Graphite`.

**Comment by timon-schelling** [comment]
@alerque Hey, Help with the aur package would be greatly appreciated. You could get started with that anytime you want. Graphite desktop Linux is in a good state. We already have a open nixpkgs PR that might be helpful to get started and identify dependencies.

Graphite repo has a flake (with desktop app packaged) under .nix. `nix run github: GraphiteEditor/Graphite?dir=.nix`

Regarding wording, although AppStores is not technically correct, I'm fine with using that terminology as it might be more meaningful for non-linux people.

**Comment by alerque** [comment]
> Hey, Help with the aur package would be greatly appreciated. You could get started with that anytime you want.

Read my message again. It's already done. It already works. It's currently only a VCS package that builds from Git HEAD at the time people install it because there are no release tags in the repository. As soon as there are I'll be making an official package, so there is no need for a 'stable' edition of the AUR packaging.

**Comment by timon-schelling** [comment]
> > Hey, Help with the aur package would be greatly appreciated. You could get started with that anytime you want.
> 
> Read my message again. It's already done. It already works. 

Sorry, I confused your aur package with a outdated one that I saw a couple weeks ago that was building the old tauri based app (very outdated).

Took a quick look at the PKGBUILD.
Looks good in general. Two comments:
1. We don't "mix and match" toolchains. The loocked nightly rust toolchain should only be used for building wgsl shaders (rustgpu requirement, just text files that get included as a string). All other components should be built with stable rust toolchain. If not having two rust toolchains is a hard requirement you could instead use the prebuilt shaders we provide (see nixpkgs PR). 
2. I'm not sure that including graphene-cli in the same package is a good idea.


The first versioned release is planned for January.

**Comment by tbaumann** [comment]
> Graphite repo has a flake (with desktop app packaged) under .nix. `nix run github: GraphiteEditor/Graphite?dir=.nix`

I'm glad for the nix support. But the subdir syntax ist quite cumbersome. It took me a while to research that actually.
It would be fantastic if the flake.nix file was moved to the repository root.
